scripts/serve_header/README.md

```diff
@@ -0,0 +1,81 @@
+serve_header.py
+===============
+
+Serves the `single_include/nlohmann/json.hpp` header file over HTTP(S).
+
+The header file is automatically amalgamated on demand.
+
+## Prerequisites
+
+> **Note:** All file locations are relative to the project root folder and all commands are assumed to be executed from the same root folder.
+
+1. Make sure these Python packages are installed.
+    ```
+    PyYAML
+    watchdog
+    ```
+
+2. To serve the header over HTTPS (which is required by Compiler Explorer at this time), a certificate is needed.
+   The recommended method for creating a locally-trusted certificate is to use [`mkcert`](https://github.com/FiloSottile/mkcert).
+   - Install the `mkcert` certificate authority into your trust store(s):
+     ```
+     $ mkcert -install
+     ```
+   - Create a certificate for `localhost`:
+     ```
+     $ mkcert localhost
```

Comment:
Could we add a certificate to the repository?

Reply:
No. You still need the CA, which also needs to be installed into trust stores. That is all made easy by `mkcert` but is otherwise quite tedious.

Also, the CA private key **must** not be shared, as it would allow anyone to eavesdrop on your secure connections.

Reply:
I know - but in the end: if HTTP without S would be allowed, we would use it for ease of use. So I just hoped to streamline the setup as much as possible. I will try on macOS later on.

Reply:
HTTP isn't allowed by Compiler Explorer. So, HTTP support is there, but not at all useful for now.

Reply:
I know - what I wanted to say: If Compiler Explorer would support HTTP, you would have used it. But since it requires HTTPS, we need certificates and all.